Assignment 2.Rmd

```diff
@@ -87,13 +98,87 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+
+EDGE.student<-EDGE%>%group_by(to)%>%summarize(sum(count))
+names(EDGE.student)<-c("id","num_dis")
+EDGE.student
+EDGE.teacher<-EDGE%>%group_by(from)%>%summarize(sum(count))
+names(EDGE.teacher)<-c("id","num_dis")
+EDGE.teacher
+VERTEX2 <- bind_rows(EDGE.student, EDGE.teacher)
+VERTEX2
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+g2
+
+plot(g2,layout=layout.fruchterman.reingold, edge.width=VERTEX2$num_dis)
+
+```
+
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+hudk4050_classes<-read.csv("/Users/liuxiaojue/Documents/2019datamining/assignment2/hudk4050-classes.csv")
+library(tidyr)
+h1 <- hudk4050_classes%>% unite(Name, First.Name, Last.Name, sep= " ")
+h2 <- gather(h1,key="class number", value="class",2:7)
+
+h2 <- na.omit(h2)
+h2 = select(h2, -c("class number"))
+h2$class <- gsub("[[:space:]]", "", h2$class)
+h2 <- h2%>%filter(Name!='ZIMO CHEN')
+h2 <- h2%>%filter(class!="HUDK4050")
+h2$count=1
+
+length.h2<-length(h2[,1])
+h2.new<-h2
+h2.new
+n=0
+for (i in 1:length.h2) {
+  
+if (h2[i,2]=="") { 
+   h2.new<-h2.new[-i+n,]
+   n=n+1
+
+}
+}
+
+h3 <- spread(h2.new,class,count)
+h3
+h3[is.na(h3)] <- 0
+#h3 is the person-class matrix
+h4 <- h3[,2:59]
+h4 <- t(t(h4)) %*% t(h4)
+colnames(h4) <- h3$Name
+rownames(h4) <- h3$Name
+diag(h4)<- NA
+h4
+#h4 is the person-person matrix
+library(igraph)
+h <- graph_from_adjacency_matrix(h4, mode="directed")
+h
+plot(h,layout=layout.fruchterman.reingold,vertex.size=5,
+     vertex.label.dist=0.5, vertex.color="pink",vertex.label.color="black" ,edge.arrow.size=0.1)
+
+betweenness.h<-betweenness(h)
+betweenness.h
+which(betweenness.h==max(betweenness.h))
+
+degree.h<-degree(h)
+degree.h
+which(degree.h==max(degree.h))
+
+# We can conclude that Yujun Zhang is the most central person in the class.
```

Comment:
Overall, good job! Try to interpret the centrality measures within the context of this dataset.